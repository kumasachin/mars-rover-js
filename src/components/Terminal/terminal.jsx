import React from 'react';
import './terminal.css';

export const Terminal = ({ printLogs }) => {
  const logs = printLogs
    ? printLogs.map((print, index) => {
        const lostClass = print.isLost ? 'Lost' : 'moving';
        const onTerminal =
          index === 0 ? 'one-robot-latest-log' : 'one-robot-log';

        return print.currentCoordinate ? (
          <div
            key={`terminal-${index}`}
            className={`${lostClass} ${onTerminal}`}
          >
            >{print.name}
            <br />>{print.currentCoordinate}
            <br />>{print.instruction}
            <br />>{lostClass}
            <br />
            ---------------------------------------
          </div>
        ) : (
          <div className={index === 0 ? 'one-map-latest-log' : 'one-map-log'}>
            >
          </div>
        );
      })
    : null;
  return (
    <div className="console">
      <header>
        <h1 data-testid="application-heading">Mars Rover project</h1>
      </header>
      <div className="consolebody">
        <p>></p>
        {logs}
      </div>
    </div>
  );
};

export default Terminal;
