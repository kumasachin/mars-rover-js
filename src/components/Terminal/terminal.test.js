import { render, cleanup } from '@testing-library/react';
import React from 'react';
import Terminal from './terminal';

describe('Terminal snapshot', () => {
  afterEach(cleanup);

  it('should take a snapshot', () => {
    const { asFragment } = render(<Terminal />);
    expect(asFragment(<Terminal />)).toMatchSnapshot();
  });
});
