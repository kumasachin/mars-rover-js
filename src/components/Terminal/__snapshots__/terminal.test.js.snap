// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terminal snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="console"
  >
    <header>
      <h1
        data-testid="application-heading"
      >
        Mars Rover project
      </h1>
    </header>
    <div
      class="consolebody"
    >
      <p>
        &gt;
      </p>
    </div>
  </div>
</DocumentFragment>
`;
